com.unity.uiwidgets/Runtime/Plugins/Android/UIWidgetsViewController.java

```diff
@@ -0,0 +1,216 @@
+    ZeroSides calculateShouldZeroSides(View unityView) {
+        Activity activity = UnityPlayer.currentActivity;
+        int orientation = activity.getResources().getConfiguration().orientation;
+        int rotation = activity.getWindowManager().getDefaultDisplay().getRotation();
+        
+        if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
+            if (rotation == Surface.ROTATION_90) {
+                return ZeroSides.RIGHT;
+            }
+            else if (rotation == Surface.ROTATION_270) {
+                return Build.VERSION.SDK_INT >= 23 ? ZeroSides.LEFT : ZeroSides.RIGHT;
+            } else if (rotation == Surface.ROTATION_0 || rotation == Surface.ROTATION_180) {
+                return ZeroSides.BOTH;
+            }
+        }
+        
+        return ZeroSides.NONE;
+    }
+    
+    public boolean lastStatusBarHidden = true;
+    
+    private int calculateBottomKeyboardInset(Rect insets) {
+        if (keyboardOpen) {
+            return insets.bottom;
+        } else {
+            return 0;
+        }
+    }
+    
+    
+    private int getNavigationBarHeight() {
+         Resources resources = UnityPlayer.currentActivity.getResources();
+         int resourceId = resources.getIdentifier("navigation_bar_height", "dimen", "android");
+         if (resourceId > 0)
+         {
+            return hasNavigationBar() ? resources.getDimensionPixelSize(resourceId) : 0;
+         }
+         
+         return 0;
+    }
+            
+    private boolean hasNavigationBar() {
+        boolean hasBar = false;
+        Resources resources = UnityPlayer.currentActivity.getResources();
+        int id = resources.getIdentifier("config_showNavigationBar", "bool", "android");
+        if (id > 0) {
+            hasBar = resources.getBoolean(id);
+        }
+        try {
+                Class systemPropertiesClass = Class.forName("android.os.SystemProperties");
+                Method m = systemPropertiesClass.getMethod("get", String.class);
+                String navBarOverride = (String) m.invoke(systemPropertiesClass, "qemu.hw.mainkeys");
+                if ("1".equals(navBarOverride)) {
+                    hasBar = false;
+                } else if ("0".equals(navBarOverride)) {
+                    hasBar = true;
+                }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        //Log.i("UIWidgetsDebug", " hasBar: " + hasBar);
+        return hasBar;
+    }
+    
+    public void updateViewMetrics() {
+        final View unityView = ((ViewGroup)UnityPlayer.currentActivity.findViewById(android.R.id.content)).getChildAt(0);
+        Rect rect = new Rect();
+        unityView.getWindowVisibleDisplayFrame(rect);
+        
+        //Log.i("UIWidgetsDebug", "calculation: " + unityView.getRootView().getHeight() + " " + rect.bottom + " " + rect.top);
+        
+        rect.bottom = unityView.getRootView().getHeight() - (rect.bottom - rect.top) - rect.top;
```

Comment:
so this line really works properly ?

Reply:
Yes. Should I fix it with  `rect.bottom = unityView.getRootView().getHeight() - rect.bottom;`?

Reply:
just keep it for now.